bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/ArithmeticGroupFunction.java

```diff
@@ -511,4 +511,69 @@ public State getStateAs(Set<Item> items, Class<? extends State> stateClass) {
         }
     }
 
+    /**
+     * This does a threshold operation. If at least one item is over 'upperLimit',

```

Comment:
This comment uses `upperState` and `lowerState` while the code below uses `activeState` and `passiveState`. I’d recommend sticking to either variant (I think I prefer upper/lower).